api/server/volume_test.go

```diff
@@ -1,110 +1,88 @@
+	assert.NoError(t, err)
 
 	// Setup request
 	name := "myvol"
 	size := uint64(1234)
 	req := &api.VolumeCreateRequest{
 		Locator: &api.VolumeLocator{Name: name},
 		Source:  &api.Source{},
-		Spec:    &api.VolumeSpec{Size: size},
+		Spec: &api.VolumeSpec{
+			HaLevel: 1,
+			Size:    size,
+		},
 	}
 
-	// Setup mock functions
-	id := "myid"
-	testVolDriver.MockDriver().
-		EXPECT().
-		Create(req.GetLocator(), req.GetSource(), req.GetSpec()).
-		Return(id, nil)
-
 	// create a volume client
 	driverclient := volumeclient.VolumeDriver(cl)
 	res, err := driverclient.Create(req.GetLocator(), req.GetSource(), req.GetSpec())
 	assert.Nil(t, err)
-	assert.Equal(t, id, res)
-
-	// Create a new global test cluster
-	tc := newTestCluster(t)
-	defer tc.Finish()
-
-	// Mock cluster
-	nodeId := "nodeid"
-	tc.MockCluster().
-		EXPECT().
-		GetNodeIdFromIp("192.168.1.1").
-		Return(nodeId, nil).Times(1)
-
-	// create a volume client with Replica IPs
-	rset := &api.ReplicaSet{Nodes: []string{"192.168.1.1"}}
-	req.Spec.ReplicaSet = rset
-	expectedSpec := req.Spec.Copy()
-	expectedSpec.ReplicaSet.Nodes = []string{nodeId}
-
-	testVolDriver.MockDriver().
-		EXPECT().
-		Create(req.GetLocator(), req.GetSource(), expectedSpec).
-		Return(id, nil)
+	assert.NotEmpty(t, res)
 
-	res, err = driverclient.Create(req.GetLocator(), req.GetSource(), req.GetSpec())
-	assert.Nil(t, err)
-	assert.Equal(t, id, res)
 }
 
-func TestVolumeCreateFailed(t *testing.T) {
+func TestVolumeCreateFailedToAuthenticate(t *testing.T) {
 
 	var err error
 
-	ts, testVolDriver := testRestServer(t)
-
+	// Setup volume rest functions server
+	ts, testVolDriver := testRestServerSdk(t)
 	defer ts.Close()
 	defer testVolDriver.Stop()
 
-	client, err := volumeclient.NewDriverClient(ts.URL, mockDriverName, version, mockDriverName)
+	// get token
+	token, err := createToken("test", "admin", "badsecret")
+	assert.NoError(t, err)
 
+	client, err := volumeclient.NewAuthDriverClient(ts.URL, mockDriverName, version, token, "", mockDriverName)
 	assert.Nil(t, err)
 	assert.NotNil(t, client)
 
-	req := &api.VolumeCreateRequest{}
-
-	// Setup mock functions
-	testVolDriver.MockDriver().
-		EXPECT().
-		Create(req.GetLocator(), req.GetSource(), req.GetSpec()).
-		Return("", fmt.Errorf("error in create"))
+	// Setup request
+	name := "myvol"
+	size := uint64(1234)
+	req := &api.VolumeCreateRequest{
+		Locator: &api.VolumeLocator{Name: name},
+		Source:  &api.Source{},
+		Spec: &api.VolumeSpec{
+			HaLevel: 1,
+			Size:    size,
+		},
+	}
 
 	// create a volume client
 	driverclient := volumeclient.VolumeDriver(client)
-
-	res, err := driverclient.Create(req.GetLocator(), req.GetSource(), req.GetSpec())
-
-	assert.NotNil(t, err)
-	assert.EqualValues(t, "", res)
-	assert.Contains(t, err.Error(), "error in create")
+	_, err = driverclient.Create(req.GetLocator(), req.GetSource(), req.GetSpec())
+	assert.Error(t, err)
```

Comment:
I need to add here a call to the SDK and get back information about the volume to assert that it is equal to what was requested.